packages/react-button/src/components/Button/useButtonState.ts

```diff
@@ -32,5 +33,17 @@ export const useButtonState = (draftState: ButtonState) => {
     }
   }
 
-  draftState.disabled = draftState['aria-disabled'] = draftState.disabled || draftState.loading;
+  draftState['aria-disabled'] = draftState.disabled;
+  draftState.disabled = draftState['aria-disabled'] && !draftState.disabledFocusable;
+
+  // Disallow keyboard events when component is disabled and eat events when disabledFocusable is set to true.
+  draftState.onKeyDown = (ev: React.KeyboardEvent<HTMLElement>) => {
+    const keyCode = getCode(ev);
+    if (draftState['aria-disabled'] && (keyCode === EnterKey || keyCode === SpacebarKey)) {
+      ev.preventDefault();
+      ev.stopPropagation();
+    } else {
+      draftState.onKeyDown?.(ev);
```

Comment:
Is it assignment correct? It will call itself recursively:

![image](https://user-images.githubusercontent.com/14183168/104297308-0427b880-54c3-11eb-8aef-52835fe630ee.png)

Reply:
That's true, I don't know what I was thinking here.